WooCommerce/Classes/ViewRelated/Orders/Order Creation/NewOrderViewModel.swift

```diff
@@ -130,6 +130,19 @@ final class NewOrderViewModel: ObservableObject {
     ///
     @Published private(set) var customerDataViewModel: CustomerDataViewModel = .init(billingAddress: nil, shippingAddress: nil)
 
+    /// View model for the customer details address form.
+    ///
+    var addressFormViewModel: CreateOrderAddressFormViewModel {
+        CreateOrderAddressFormViewModel(siteID: siteID,
+                                        addressData: .init(billingAddress: orderSynchronizer.order.billingAddress,
+                                                           shippingAddress: orderSynchronizer.order.shippingAddress),
+                                        onAddressUpdate: { [weak self] updatedAddressData in
+            let input = Self.createAddressesInput(from: updatedAddressData)
+            self?.orderSynchronizer.setAddresses.send(input)
+            self?.trackCustomerDetailsAdded()
+        })
+    }
```

Comment:
I tested the feature and it seems to work but I can see that now this view model is not being retained, right?

Should we make this a lazy var to be on the safe side?

Reply:
Oh, you're right. But making this a lazy var doesn't work as-is because then what we pass in for its `addressData` isn't ever updated. Handling that takes us back to something more like how it was done in [7626e15](https://github.com/woocommerce/woocommerce-ios/pull/6534/commits/7626e15d3db238d837e898ae6fb089818f755e3d), so it seems I was mistaken about this being a better approach. 😬 

I think I'll revert these latest changes and address the comments from [your previous review](https://github.com/woocommerce/woocommerce-ios/pull/6534#pullrequestreview-932565888) + the UI issue you found here.